From 11f1ae429fdcc5553ed3f3de32c84eeebeea17b2 Mon Sep 17 00:00:00 2001
From: Shreyas NC <shreyas.nc@intel.com>
Date: Tue, 30 May 2017 19:37:19 +0530
Subject: [PATCH 0735/2278] ASoC: Intel: Skylake: Fix incorrect parsing of pipe
 tokens

To avoid parsing of pipe related tokens multiple times for a case where
the pipe has more than one module, a logic was added to parse these
tokens only once. But, the existing logic would parse these
tokens only if there are more than one module in the pipe. So, for a
pipe with single module, the logic is insufficient.

So, fix it by updating the flag accordingly.

Change-Id: Ie183f14eaf98f21cf87691e0e681b77342706b37
Signed-off-by: Shreyas NC <shreyas.nc@intel.com>
Signed-off-by: Diwakar, Praveen <praveen.diwakar@intel.com>
Reviewed-by: audio_build <audio_build@intel.com>
Reviewed-by: Prodduvaka, Leoni <leoni.prodduvaka@intel.com>
Reviewed-by: S, Pavan K <pavan.k.s@intel.com>
Reviewed-by: R, Dharageswari <dharageswari.r@intel.com>
Reviewed-by: Singh, Guneshwor O <guneshwor.o.singh@intel.com>
Reviewed-by: Kale, Sanyog R <sanyog.r.kale@intel.com>
Reviewed-by: Kp, Jeeja <jeeja.kp@intel.com>
Tested-by: Sm, Bhadur A <bhadur.a.sm@intel.com>
---
 sound/soc/intel/skylake/skl-topology.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index 1c46578..70b5870 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -2836,6 +2836,7 @@ static int skl_tplg_get_token(struct device *dev,
 			}
 			return is_pipe_exists;
 		}
+		is_pipe_exists = 0;
 
 		break;
 
@@ -2849,7 +2850,7 @@ static int skl_tplg_get_token(struct device *dev,
 	case SKL_TKN_U32_PMODE:
 	case SKL_TKN_U32_PIPE_DIRECTION:
 	case SKL_TKN_U32_NUM_CONFIGS:
-		if (is_pipe_exists) {
+		if (!is_pipe_exists) {
 			ret = skl_tplg_fill_pipe_tkn(dev, mconfig->pipe,
 					tkn_elem->token, tkn_elem->value);
 			if (ret < 0)
-- 
2.7.4

